composables/use-issues-by-column.ts

```diff
@@ -0,0 +1,40 @@
+import type { ColumnTitle } from '~/constants/columns'
+import type { IssueStatus } from '~/constants/issues'
+import type { Issue } from '~/types/issues'
+
+function filterIssuesWithoutStatus(issues: Issue[]): Issue[] {
+  const issuesWithoutStatus = issues.filter(
+    (issue) =>
+      !issue.labels.some((label) => label.startsWith(createPartialDataLabel('status'))),
+  )
+
+  return issuesWithoutStatus
+}
+
+function filterIssuesByStatus(issues: Issue[], status: IssueStatus): Issue[] {
+  const issuesWithStatus = issues.filter((issue) =>
+    issue.labels.includes(createDataLabel('status', status)),
+  )
+
+  return issuesWithStatus
+}
+
+function useIssuesByColumn(): Record<ColumnTitle, ComputedRef<Issue[]>> {
+  const route = useRoute('node-rid')
+  const response = useHttpdFetch('/projects/{rid}/issues', {
+    path: {
+      rid: route.params.rid,
+    },
+  })
+
+  const issues = response.data as Ref<Issue[] | undefined>
+
+  return {
+    'not-planned': computed(() => filterIssuesWithoutStatus(issues.value ?? [])),
+    'todo': computed(() => filterIssuesByStatus(issues.value ?? [], 'todo')),
+    'doing': computed(() => filterIssuesByStatus(issues.value ?? [], 'doing')),
```

Comment:
Do all those need to be computed? If anything we should see if they should be [`shallowRef()`](https://vuejs.org/api/reactivity-advanced.html#shallowref) because tracking (and especially setting up tracking!) every prop of every array item can't be too efficient, especially if we don't need it.

Reply:
I wasn't very sure this was right in the first place TBH.

I definitely need to do a deeper dive on Vue's reactivity once we have more breathing room. Thanks for helping me out!

This composable has been removed in favor of a simple function that transforms the data as we need it.